### PR TITLE
Refactor copy domain

### PR DESCRIPTION
**Pull request by TylerSheffels**
http://manage.dimagi.com/default.asp?186025

I'm not particularly proud of this, and setting it to run in a single process seems to increase the run time a lot. Maybe I should investigate using gevent to spin up new threads in the event I cant spin up new processes? @czue 

CC: @gcapalbo

### CODE REVIEW

File: corehq/apps/domainsync/management/commands/copy_domain.py [hunk 1]
```diff
@@ -169,23 +176,28 @@ def copy_docs(self, sourcedb, domain, simulate, startkey=None, endkey=None, doc_
...
 
         err_log = self._get_err_log()
 
-        queue = Queue(150)
-        for i in range(NUM_PROCESSES):
-            Worker(queue, sourcedb, self.targetdb, exclude_types, total, simulate, err_log, exclude_attachments).start()
+        if self.run_multi_process:
+            queue = Queue(150)
+            for i in range(NUM_PROCESSES):
+                Worker(queue, sourcedb, self.targetdb, exclude_types, total, simulate, err_log, exclude_attachments).start()
```
Comment (dimagimon) [inline corehq/apps/domainsync/management/commands/copy_domain.py]: E501 line too long (124 > 115 characters)

**Comment by gcapalbo** [comment]
has this been tested / are there automated tests that would catch if anything broke here?

**Comment by TylerSheffels** [comment]
@gcapalbo There are not automated tests for this, I tested it both with and without the --dont-run-multi-process flag.

**Comment by czue** [comment]
looks good to me, and is enough of a one-off that I'm ok merging.

you could also look into the `couch_migrations` framework that @dannyroberts and @snopoke made as a replacement for this

File: corehq/apps/domainsync/management/commands/copy_domain.py [hunk 2]
```diff
@@ -82,7 +82,13 @@ class Command(BaseCommand):
                     dest='postgres_password',
                     default='',
                     help="Password for postgres database to pull additional data from. If not specified will "
-                         "default to the value in settings.DATABASES")
+                         "default to the value in settings.DATABASES"),
+        make_option('--dont-run-multi-process',
+                    action='store_false',
+                    dest='run_multi_process',
+                    default=True,
+                    help="If set to true this spawn multiple processes which should speed up the time taken to "
```
Comment (dannyroberts) [inline corehq/apps/domainsync/management/commands/copy_domain.py]: `spawns`

**Pull request merged**